src/test_helpers.rs

```diff
@@ -101,14 +116,15 @@ pub fn test_config_with_port_xds_addr_and_root_cert(
         // Switch all addressed to localhost (so we don't make a bunch of ports expose on public internet when someone runs a test),
         // and port 0 (to avoid port conflicts)
         // inbound_addr cannot do localhost since we abuse that its listening on all of 127.0.0.0/8 range.
-        inbound_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), 0),
+        inbound_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), hbone_port),
         socks5_addr: Some(SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0)),
         admin_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0),
         readiness_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0),
         stats_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0),
         outbound_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0),
         inbound_plaintext_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0),
         dns_proxy_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0),
+        illegal_ports: HashSet::new(), // for "direct" tests, since the ports are latebound, we can't test illegal ports
```

Comment:
If we are not setting this, do we need to fix inbound_addr to a non-zero port?

The current approach will flake if you run multiple tests concurrently like `cargo-nextest` or others (I run it to detect flakes, etc)

Reply:
Ah we need the hbone port

Reply:
How bad is it to pass the hbone port still but keep the illegal_ports?

Reply:
Sorry I feel a bit like https://xkcd.com/1172/...

Reply:
Still gross, but gross in one signposted spot. Should do for `nextest` and friends. Does for me.